pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -271,6 +268,16 @@ func (f *ring1Factory) LogsForObject(object, options runtime.Object) (*restclien
 		}
 		return nil, fmt.Errorf("cannot get the logs from %v", gvks[0])
 	}
+
+	sortBy := func(pods []*v1.Pod) sort.Interface { return controller.ByLogging(pods) }
+	pod, numPods, err := GetFirstPod(clientset.Core(), namespace, selector, 20*time.Second, sortBy)
```

Comment:
The same applies for `AttachablePodForObject`, I've created https://github.com/kubernetes/kubernetes/issues/41786 to track it.